### ISSUE TITLE
[Bug] Incorrect Gen 1 Hue Dimmer Switch Button Mapping in HomebridgeZ2M (DOWN-OFF-ON-UP)

### ISSUE THREAD

**Issue by perfecblue** [itavero/homebridge-z2m#155]
Please use the template and fill in all the data requested below!

**Describe the bug**
Hi, I'm trying to use the Gen 1 Hue Dimmer (Philips 324131092621) with a CC2652p-based Zigbee Gateway on Homebridge in hopes of getting the LONG PRESS functionality that isn't provided by the Hue Bridge itself in Homekit.  

The Dimmer Switch successfully paired (with the "legacy: true" default config) was able to be seen and operated in Homekit. However, I realized quickly that the button mapping of the Dimmer when recognized by the HomebridgeZ2M plugin was as follows:

Button 1 (down):
SINGLE: down-press
LONG : down-hold

Button 2 (off):
SINGLE: off-press
LONG : off-hold

Button 3 (on):
SINGLE: on-press
LONG : on-hold

Button 4 (up):
SINGLE: up-press
LONG : up-hold

This seems odd to me, since the actual button placement on the remote from top to bottom is ON - UP - DOWN - OFF. I understand the support of this device was added a long time ago, but is there anyway I can change the button mapping when it is brought into Homebridge?

p.s. From Zigbee2MQTT's frontend, the button mapping is correct, it is only after it's brought into Homekit that the mapping changes.

Thank you very much!

**Related devices**
Gen 1 Philips Hue Dimmer Switch (Philips 324131092621)

**To Reproduce**
Standard install with Device type specific configuration set to "legacy: true" or just leave it blank as default.

**Expected behavior**
The actual button mapping from top to bottom should be: ON - UP - DOWN - OFF.

**MQTT messages**
Pressing the buttons in order of:
ON - short press
ON - long press
UP - short press
UP - long press
DOWN - short press
DOWN - long press
OFF - short press
OFF - long press

`MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"on-press","battery":100,"brightness":33,"counter":1,"duration":0,"linkquality":117,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"on-hold","battery":100,"brightness":33,"counter":1,"duration":0.801,"linkquality":105,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"on-hold-release","battery":100,"brightness":33,"counter":1,"duration":1.3,"linkquality":102,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"up-press","battery":100,"brightness":65,"counter":1,"duration":0,"linkquality":120,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"up-hold","battery":100,"brightness":97,"counter":1,"duration":0.801,"linkquality":123,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"up-hold-release","battery":100,"brightness":97,"counter":1,"duration":1.571,"linkquality":105,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"down-press","battery":100,"brightness":65,"counter":1,"duration":0,"linkquality":123,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"down-hold","battery":100,"brightness":33,"counter":1,"duration":0.805,"linkquality":117,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"down-hold-release","battery":100,"brightness":33,"counter":1,"duration":1.547,"linkquality":120,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"off-press","battery":100,"brightness":33,"counter":1,"duration":0,"linkquality":120,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"off-hold","battery":100,"brightness":33,"counter":1,"duration":0.802,"linkquality":120,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/邊桌 Hue Dimmer', payload '{"action":"off-hold-release","battery":100,"brightness":33,"counter":1,"duration":1.363,"linkquality":117,"update":{"state":"idle"},"update_available":false}'`

**Versions used**
 - This plugin: v1.1.3
 - Homebridge: v1.3.4
 - Zigbee2MQTT: v1.18.2
 - Homebridge Config UI X: v4.41.1

**Comment by itavero** [comment]
The button mapping is shown in the Homebridge log upon startup.
Does it match this mapping or not?

The reason it might not map to the physical position on the device (or even the buttons on the device), is because this information is not known in Zigbee2MQTT and therefor not known in this Homebridge plugin.
In reality without this information being provided by Zigbee2MQTT, the plugin cannot do this "correctly" without a lot of configuration (which is what I want to avoid).

**Comment by perfecblue** [comment]
Hi, yes, the button mapping is exactly as shown in the homebridge log as button 1 (down) / button 2 (off) / button 3 (on) / button 4 (up) and it's how it was presented in Homekit as a result.  

I see, I guess since ZigBee2MQTT has no idea that the ON button stands for Button 1 and the UP button stands for Button 2 and so on, I guess it's not possible to have it at the correct physical positions at this point?

**Comment by itavero** [comment]
It is not possible. We could potentially create a configuration for this, but I personally don't see a use case for it and it would only complicate the usability. If you really want this, please open up a feature request issue describing what this feature should look like.

Functionally, it doesn't change a thing and HomeKit still won't show labels that directly relate to what's on the physical device (numbers as opposed to "On", "Off", "Up" and "Down").
Next to that, HomeKit will still show which event was triggered when you are in the screen to configure the actions in the Home app.
In other words, if you perform the action on the remote for which you want to configure the action in HomeKit, it should give you an indication of that in the Home app.

Closing this issue, as the device is working as defined.
